cpp/include/cudf/io/writers.hpp

```diff
@@ -209,8 +209,15 @@ class writer {
    *
    * @param[in] pq_chunked_state State information that crosses _begin() / write_chunked() / _end()
    * boundaries.
+   * @param[in] return_filemetadata If true, return the raw file metadata
+   * @param[in] metadata_out_file_path Column chunks file path to be set in the raw output metadata
+   *
+   * @return A parquet-compatible blob that contains the data for all rowgroups in the list
    */
-  void write_chunked_end(struct pq_chunked_state& state);
+  std::unique_ptr<std::vector<uint8_t>> write_chunked_end(
```

Comment:
why return unique_ptr of container instead of container itself?

Reply:
My C++/cython brain is pretty rusty, so I'm not completely sure - I was simply following the approach used for the non-chunked parquet writer.  My understanding is that the unique pointer allows us to avoid any data copying in cython (is this correct @kkraus14 ?)

Reply:
> My C++/cython brain is pretty rusty, so I'm not completely sure - I was simply following the approach used for the non-chunked parquet writer. My understanding is that the unique pointer allows us to avoid any data copying in cython (is this correct @kkraus14 ?)

Maybe @brandon-b-miller or @shwina know, but I vaguely remember it allowing us to more easily wrap the vector in a Cython class. It could have just been our naivety surrounding C++ containers though 😄